include/inexor/vulkan-renderer/wrappers/instance.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/availability_checks.hpp"
+
+#include <spdlog/spdlog.h>
+#include <vulkan/vulkan.h>
+
+#include <cassert>
+#include <memory>
+#include <optional>
+#include <string>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+/// @brief A RAII wrapper class for VkInstance.
+/// @note The instantiation of this class must be synchronized externally.
+class Instance {
+protected:
+    VkInstance instance = VK_NULL_HANDLE;
+    std::unique_ptr<AvailabilityChecksManager> availability_checks = std::make_unique<AvailabilityChecksManager>();
```

Comment:
Does this need to be a `unique_ptr` if it gets initialized inline anyway?

Reply:
Right. Not really. I will make it a normal member.

Reply:
done.